modeling/datasets_sc.py

```diff
@@ -0,0 +1,369 @@
+                    gt_subvolumes = volume2subvolumes(volume=gt, subvolume_size=self.subvolume_size, stride_size=self.subvolume_size)
+                    assert len(sag_img_subvolumes) == len(gt_subvolumes)
+
+                    # Collect individual subvolume predictions for full volume segmentation (i.e. full scan for one subject)
+                    pred_subvolumes = []
+                    for i in range(len(sag_img_subvolumes)):
+                        sag_img_subvolume, gt_subvolume = sag_img_subvolumes[i], gt_subvolumes[i]
+                        # Normalize images to zero mean and unit variance
+                        normalize_instance = NormalizeInstance()
+                        if sag_img_subvolume.std() < 1e-5:
+                            # If subvolumes uniform: do mean-subtraction
+                            sag_img_subvolume = sag_img_subvolume - sag_img_subvolume.mean()
+                        else:
+                            sag_img_subvolume, _ = normalize_instance(sample=sag_img_subvolume, metadata={})
+                        # twice unsqueezed such that batch_size=1 and num_channels=1
+                        x1 = torch.tensor(sag_img_subvolume, dtype=torch.float).view(1, 1, *sag_img_subvolume.shape)
+
+                        # Get the standard subvolume prediction
+                        seg_y_hat = model(x1).squeeze().detach().cpu().numpy()
+
+                        pred_subvolumes.append(seg_y_hat)
+
+                    # Convert the list of subvolume predictions to a single volume segmentation / prediction
+                    pred = subvolumes2volume(subvolumes=pred_subvolumes, volume_size=adjusted_center_crop_size)
+                    assert pred.shape == gt.shape
+
+                # Apply the original center-crop size in case it was adjusted before
+                if pred.shape != self.center_crop_size:
+                    gt_sum_before_crop = np.sum(gt)
+                    center_crop = CenterCrop(size=self.center_crop_size)
+                    pred = center_crop(sample=pred, metadata={'crop_params': {}})[0]
+                    gt = center_crop(sample=gt, metadata={'crop_params': {}})[0]
+
+                    # Check if padding & un-padding removes any lesion GTs; only continue if it does not
+                    if abs(np.sum(gt) - gt_sum_before_crop) > 1e-6:
+                        # NOTE: Apparently np.sum() can have epsilon differences even with same values!
+                        raise ValueError('Padding & un-padding cropped out lesions! Check your center-crop parameters!')
+            
+                # Calculate the test metrics on the "soft" prediction only
+                for metric_fn in self.metric_fns:
+                    res = metric_fn(pred, gt)
+                    dict_key = metric_fn.__name__
+                    self.test_metrics[dict_key].append(res)
```

Comment:
Some scores were NaN (we should also investigate why) so we can remove them in performance computation with:
```suggestion
                    if dict_key == 'dice_score' and np.isnan(res):
                        continue

                    self.test_metrics[dict_key].append(res)
```